src/common.fypp

```diff
@@ -156,4 +156,61 @@ ${prefix + joinstr.join([line.strip() for line in txt.split("\n")]) + suffix}$
   #:endcall
 #:enddef
 
+#!
+#! Generates an array rank suffix for subarrays along a dimension
+#!
+#! Args:
+#!   varname (str): Name of the variable to be used as origin
+#!   rank (int): Rank of the original variable
+#!   dim (int): Dimension of the variable
+#!
+#! Returns:
+#!   Array rank suffix string enclosed in braces
+#!
+#! E.g.,
+#!  select_subvector('j', 5, 2)
+#!   -> (j_, :, j___, j____, j_____)
+#!
+#:def select_subvector(varname, rank, idim)
+  #:assert rank > 0
+  #:call join_lines(joinstr=", ", prefix="(", suffix=")")
+    #:for i in range(1, idim)
+      ${varname}$${ "_" * (i) }$
```

Comment:
I recall in one of the previous stats modules, we commented that using integer indexes, i.e. (`j1`, `j2`, `j3`, and so on), seemed more legible than the use of multiple underscores.

Is this applicable here too?

Reply:
You are right. I didn't remember this discussion. I'll change that.